crates/threshold-bls/src/poly.rs

```diff
@@ -56,74 +67,63 @@ impl<C: Element> Poly<C> {
     pub fn zero() -> Self {
         Self::from(vec![C::zero()])
     }
+
+    fn is_zero(&self) -> bool {
+        self.0.is_empty() || self.0.iter().all(|coeff| coeff == &C::zero())
+    }
+
+    /// Performs polynomial addition in place
+    pub fn add(&mut self, other: &Self) {
+        // if we have a smaller degree we should pad with zeros
+        if self.0.len() < other.0.len() {
+            self.0.resize(other.0.len(), C::zero())
+        }
+
+        self.0.iter_mut().zip(&other.0).for_each(|(a, b)| a.add(&b))
```

Comment:
What's the behavior if `other` is smaller than `self` ? not sure for zip()

Reply:
If `other` is smaller than `self` it'll iterate until `other`'s length

Reply:
So it's not good, because the result should contain 4 as well no ? 
I would either add a comment saying both must be same length or change such that it handles this case ? (like pad the smallest of the two with 0 for example such that we can still use zip ?)

Reply:
The result is still going to be of the correct size, e.g.:

PolyA: [1,2,3]
PolyB: [4,5]

This will return [5,7,3]

Reply:
Ah sorry right you're iterating directly on the coefficient of self. My bad.